src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -204,23 +204,38 @@ public String placeUnits(final Collection<Unit> units, final Territory at, final
     if (error != null) {
       return error;
     }
-    // System.out.println("Placing " + MyFormatter.unitsToTextNoOwner(units) + " at " + at.getName() + " by " +
-    // player.getName());
     final List<Territory> producers = getAllProducers(at, m_player, units);
     producers.sort(getBestProducerComparator(at, units, m_player));
-    // System.out.println("Producers: " + producers);
     final IntegerMap<Territory> maxPlaceableMap = getMaxUnitsToBePlacedMap(units, at, m_player, true);
-    // System.out.println("Max Place Map: " + maxPlaceableMap);
-    final List<Unit> unitsLeftToPlace = new ArrayList<>(units);
-    unitsLeftToPlace.sort(getUnitConstructionComparator());
+
     // sort both producers and units so that the "to/at" territory comes first, and so that all constructions come first
     // this is because the PRODUCER for ALL CONSTRUCTIONS must be the SAME as the TERRITORY they are going into
-    for (final Territory producer : producers) {
-      if (unitsLeftToPlace.isEmpty()) {
+    final List<Unit> unitsLeftToPlace = new ArrayList<>(units);
+    unitsLeftToPlace.sort(getUnitConstructionComparator());
+
+    while (true) {
+      if (unitsLeftToPlace.isEmpty() || producers.isEmpty()) {
         break;
       }
-      // units may have special restrictions like RequiresUnits
 
+      // Get next producer territory
+      Territory producer = producers.get(0);
+      if (producers.size() > 1) {
+        producer = getRemotePlayer().selectProducerTerritoryForUnits(producers, at);
+        if (producer == null) {
+          break;
+        }
+      }
+      producers.remove(producer);
```

Comment:
The 15 lines above this comment are the core of the change. It checks if there are multiple factories and if so then call the player to choose one. It loops until all of the units are placed or all territories are utilized.